[Frontend] Start Table Component

paimasiko opened the pull request:
#48 

Table Component

paimasiko commented on src/components/BogTable/styles.module.css:
```diff
@@ -0,0 +1,100 @@
+@media (max-width: 767px) {
+  .columnHeaderCell,
+  .rowHeaderCell,
+  .cell {
```
fixed i think

paimasiko commented on src/components/BogTable/BogTable.tsx:
```diff
@@ -0,0 +1,116 @@
+import { Table, Theme } from '@radix-ui/themes';
+import React, { ReactElement, useMemo } from 'react';
```
fixed i think

Nathan-Papa commented on src/components/BogTable/BogTable.stories.tsx:
```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import type { Meta, StoryObj } from '@storybook/react-vite';
+import BogTable, { ColumnHeaderCellContent, TableRow } from './BogTable';
```
Ignore this

Nathan-Papa commented on src/components/BogTable/BogTable.stories.tsx:
```diff
@@ -0,0 +1,57 @@
+const rows: TableRow[] = [
+  {
+    cells: [
+      { content: <span>John Doe</span> },
+      { content: <span>30</span> },
+      { content: <span>USA</span> },
+      { content: <span>Developer</span> },
+    ],
+  },
+  {
+    cells: [
+      { content: <span>Jane Smith</span> },
+      { content: <span>25</span> },
+      { content: <span>Canada</span> },
+      { content: <span>Designer</span> },
+    ],
+  },
+  {
+    cells: [
+      { content: <span>Sam Wilson</span> },
+      { content: <span>35</span> },
+      { content: <span>UK</span> },
+      { content: <span>Engineering Manager</span> },
+    ],
```
Can you just make sure that normal strings work that aren't wrapped in React Elements, that would be ideal. But you can leave this or replace it, I just want to make sure plain text works and looks good.

Nathan-Papa commented on src/components/BogTable/BogTable.stories.tsx:
```diff
@@ -0,0 +1,57 @@
+  args: {
+    columnHeaders,
+    rows,
+    size: 'responsive',
```
Can you delete the size here. I want the responsive size to be the default when no prop is passed in and I want it to be reflected as such in the story.

Nathan-Papa commented on src/components/BogTable/BogTable.tsx:
```diff
@@ -0,0 +1,116 @@
+  ...rootProps
+}) => {
+  const breakpoint = useResponsive();
+  const resolvedSize =
+    size === 'responsive' ? getSizeFromBreakpoint(breakpoint) : size;
+  const radixSize = mapSizeToRadix(resolvedSize);
```
Can you just make another helper method in `src/utils/breakpoints/breakpoints.ts` called `getNumericalSizeFromBreakpoint` which basically does exactly what `getSizeFromBreakpoint` does but returns 1, 2, and 3 instead of small, medium, and large? I think that would get rid of the radix styles import and fit the existing patterns better. If you still need the radix styles import that's fine.

Nathan-Papa commented on src/components/BogTable/styles.module.css:
```diff
@@ -0,0 +1,77 @@
+@layer base {
+    .container {
+        min-height: auto !important;
+        border-radius: 8px !important;
```
I'm sorry seeing px in my stylesheets trigger me. I set the font-size globally to 10px so 1rem = 10px. Can you change the border-radius, padding, and width measurements in this file to use rem? For example this border-radius would be 0.8rem.

Nathan-Papa commented on src/components/BogTable/styles.module.css:
```diff
@@ -0,0 +1,77 @@
+
+  .size3 {
+    min-height: 4rem !important;
+    min-width: 15.53125rem !important;
```
holy precision 😂 (just making fun dw about it)

Nathan-Papa reviewed (changes requested):
just a couple small changes. it's all super nitty so dw about it, thanks for the work :)

paimasiko commented:
Fixed all the stuff mentioned and made it so that the content type is a ReactNode, so it can support plain text as well as React Elements.

I added the new method in breakpoints.ts but I still need the radix-ui styles import so that the table looks right.

Nathan-Papa reviewed (approved):
thank you!
